hibernate-core/src/main/java/org/hibernate/type/spi/TypeConfiguration.java

```diff
@@ -486,6 +487,26 @@ public Dialect getDialect() {
 					: sessionFactory.getJdbcServices().getDialect();
 		}
 
+		public org.hibernate.type.Type getIdentifierType(String className) throws MappingException {
```

Comment:
Not exactly specific to this work, but I think we should consider moving `Scope` as a top-level class as it is starting to get "large".  It was initially a very small class when I first developed it.